### PR TITLE
[XGAL-71] Add user login and profile page

### PR DESCRIPTION
**Pull request by thongphp**

### CODE REVIEW

File: app/Providers/AppServiceProvider.php [hunk 1]
```diff
@@ -69,12 +71,13 @@ public function register(): void
      */
     public function boot(): void
     {
-        Totem::auth(function ($request) {
+        Totem::auth(static function (Request $request) {
```
Comment (jooservices) [inline app/Providers/AppServiceProvider.php:74]: Why use static ?

File: app/Providers/AppServiceProvider.php [hunk 2]
```diff
@@ -69,12 +71,13 @@ public function register(): void
...
             if (!$user = $request->user()) {
                 return false;
             }
-            $authenticatedUsers = config('services.authenticated.emails');
-            return in_array($user->email, $authenticatedUsers);
+
+            /* @var User $user */
```
Comment (jooservices) [inline app/Providers/AppServiceProvider.php:79]: No need

File: resources/views/base.blade.php [hunk 3]
```diff
@@ -118,6 +118,27 @@
         </div>
     </div>
 </div>
+@guest
```
Comment (jooservices) [inline resources/views/base.blade.php:121]: It's hard code modal ? Why not blank modal and use ajax to render right content ?

File: app/Providers/AppServiceProvider.php (hunk 1 above)
Comment (thongphp) [reply app/Providers/AppServiceProvider.php:74]: Static function for closure could increase performance while running.

File: resources/views/base.blade.php (hunk 3 above)
Comment (thongphp) [reply resources/views/base.blade.php]: I already updated Modal and use as Global Modal (content will rendered dynamically)

**Pull request merged**